File: fastapi-addon/manifest.py
from __future__ import annotations


GENRES = [
    "All",
    "Talk Of The Town",
    "Watch It With District",
    "Editors Pick",
    "Netflix",
    "JioHotstar",
    "Prime",
    "Crunchyroll",
]


def _catalog(content_type: str, name: str) -> dict:
    return {
        "id": "moctale-all",
        "type": content_type,
        "name": name,
        "extra": [
            {
                "name": "genre",
                "options": GENRES,
                "isRequired": False,
            }
        ],
    }


def build_manifest(configured: bool = False) -> dict:
    return {
        "id": "org.moctale.catalog",
        "version": "1.0.0",
        "name": "Moctale Catalog",
        "description": "Movies and shows curated from moctale.in",
        "logo": "https://www.moctale.in/favicon.ico",
        "resources": ["catalog"],
        "types": ["movie", "series"],
        "catalogs": [
            _catalog("movie", "Moctale - Movies"),
            _catalog("series", "Moctale - Series"),
        ],
        "behaviorHints": {
            "adult": False,
            "configurable": True,
            "configurationRequired": not configured,
        },
        "idPrefixes": ["moctale-", "tt"],
    }

File: stremio-addon/src/manifest.js
const PORT = process.env.PORT || 7000;
const ADDON_URL = process.env.ADDON_URL || `http://127.0.0.1:${PORT}`;

export function buildManifest() {
  return {
    id: "org.moctale.catalog",
    version: "1.0.0",
    name: "Moctale Catalog",
    description: "Movies & shows curated from moctale.in",
    logo: "https://www.moctale.in/favicon.ico",
    resources: ["catalog"],
    types: ["movie", "series"],
    catalogs: [
      {
        id: "moctale-all",
        type: "movie",
        name: "Moctale - Movies",
        extra: [
          {
            name: "genre",
            options: [
              "Editors Pick",
              "Netflix",
              "JioHotstar",
              "Prime Video",
              "Hotstar",
            ],
          },
        ],
      },
      {
        id: "moctale-all",
        type: "series",
        name: "Moctale - Series",
        extra: [
          {
            name: "genre",
            options: [
              "All",
              "Talk Of The Town",
              "Watch It With District",
              "Editors Pick",
              "Netflix",
              "JioHotstar",
              "Prime Video",
              "Hotstar",
            ],
          },
        ],
      },
    ],
    behaviorHints: { adult: false },
    idPrefixes: ["moctale-"],
  };
}

export function handler({ type, id, extra }) {
  // Stremio SDK-compatible handler — returns Promise<{ metas }>
  const genre = extra?.genre || "All";
  return Promise.resolve({ metas: [] });
}
